histomicsui/web_client/vue/components/EditHeatmapOrGridData.vue

```diff
@@ -0,0 +1,229 @@
+<script>
+import tinycolor from 'tinycolor2';
+import _ from 'underscore';
+import ColorPickerInput from './ColorPickerInput.vue';
+export default {
+    props: ['elementData'],
+    emits: ['submit', 'cancel'],
+    components: {
+        ColorPickerInput
+    },
+    data() {
+        return {
+            type: this.elementData.type,
+            interpretation: this.elementData.interpretation,
+            radius: this.elementData.radius,
+            normalizeRange: this.elementData.normalizeRange,
+            colorRange: _.clone(this.elementData.colorRange),
+            colorRangeData: [],
+            rangeValues: _.clone(this.elementData.rangeValues),
+            minColor: this.elementData.minColor,
+            maxColor: this.elementData.maxColor,
+            stepped: this.elementData.stepped,
+            scaleWithZoom: this.elementData.scaleWithZoom,
+            validationErrors: []
+        };
+    },
+    computed: {
+        headerMessage() {
+            if (this.type === 'heatmap') {
+                return 'Edit Heatmap Element';
+            }
+            return 'Edit Grid Data Element';
+        }
+    },
+    methods: {
+        getColorString(color) {
+            return tinycolor(color).toString();
+        },
+        addColor(index) {
+            const newEntry = {
+                r: 0,
+                g: 0,
+                b: 0,
+                a: 0,
+                colorString: 'rgba(0, 0, 0, 0)',
+                value: 0
```

Comment:
When we add a color that is between two other colors, can we default it to a value halfway between the existing values?

Similarly, if it is a new first color, lets set the value to 0 and the color to whatever the lowest-rank color (excluding min/max( is but with a 0 opacity, and, if it is a new last color, the value should be the greater of 1 or the value of the highest-rank color record, and the color the same as the highest-rank color.  If the highest-rank opacity is 0, set it to 1.

Reply:
Take a look at dd3938a for these changes.